### DISCUSSION TITLE
Serving fonts while Martin is used with nginx returns HTTP 400

### DISCUSSION THREAD

**Discussion by pirttju** [maplibre/martin#1115] [Q&A] [answered]
I've tried to set up Martin with nginx as a reverse proxy for caching tiles and fonts.

The current situation is that the tiles are served correctly from my server. But when the maplibre client tries to request a font, nginx returns HTTP 400. I've already checked that my map styles point to correct URLs. I've also tried the font URLs locally using curl. Direct requests to Martin seem to work while request via nginx do not.

For nginx I'm using the example configuration file from https://github.com/maplibre/martin/blob/main/demo/frontend/nginx.conf with minor modifications to allow for CORS.

[logs.txt](https://github.com/maplibre/martin/files/13813232/logs.txt)

$ martin --version
martin 0.13.0

Martin debug log (related rows):
Requested via nginx:
[2024-01-02T19:25:53Z ERROR actix_http::h1::dispatcher] stream error: request parse error: invalid HTTP version specified
Requested directly:
[2024-01-02T19:26:26Z INFO  actix_web::middleware::logger] xx.xxx.xxx.xxx "GET /font/Lato%20Regular/0-255 HTTP/1.1" 200 36890 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:121.0) Gecko/20100101 Firefox/121.0" 0.021856

**Comment by nyurik** [comment, marked as answer]
This is clearly an nginx issue, and I suspect this is due to a space in the URL.  Your nginx setup doesn't seem to be escaping that space, and thus the HTTP parser is having issues parsing the request.

**Comment by catalinconstant** [comment]
Martin example configuration : 

```nginx
location ~ /tiles/(?<fwd_path>.*) {
    proxy_set_header  X-Rewrite-URL $uri;
    proxy_set_header  X-Forwarded-Host $host:$server_port;
    proxy_set_header  X-Forwarded-Proto $scheme;
    proxy_redirect    off;

    proxy_pass        http://martin:3000/$fwd_path$is_args$args;
}
```

can be rewritten as:

```nginx
location /tiles/  {
  .... 
 proxy_pass        http://martin:3000/;
} 
```

 `$fwd_path` will be unescaped  since nginx 1.15. Note that  the trailing / in proxy_pass is important.

**Reply by nyurik** [reply]
@catalinconstant thx!  Perhaps we should update the example/docs/demo site too?

**Reply by catalinconstant** [reply]
Yes, Martin example will not work with the late versions of nginx. 
Also is advisable to add in 
location  /tiles/ {
...
proxy_headers_hash_max_size 512;
proxy_headers_hash_bucket_size 128;
...
}
directives ;
To make use of http2 (all latest versions of nginx are compiled with http2 module) to add also 
server {
		listen 443 ssl http2 default_server;
....
}
If you want I can send you a complete  nginx.conf.
